### PR TITLE
[IMP] website_forum: clean public access

### PR DESCRIPTION
**Pull request by amdi-odoo**
Purpose
=======
Clean the controller routes returning the json 'anonymous_user' error. Change the auth="public" to auth="user" and handling the public users from the js before calling the routes to prevent calling useless rpcs.

Specification
=============
Implementing a js tool method that checks whether or not the user is public and if the user is, sends a warning alert saying they cannot perform the action without being logged.

Here are a list of all the routes handled:
- Toggle favourite a question
- Toggle correct a post answer
- Upvote or downvote a post
- Flag a post
- Delete a comment
- Flag a post

The toggle favourite action doesn't have any js as the route is directly called using a href. No need to change it for a js click event as the favorite button is only visible if the user is logged.

Task-4042587

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_forum/static/src/js/website_forum.js [hunk 1]
```diff
@@ -187,6 +187,20 @@ publicWidget.registry.websiteForum = publicWidget.Widget.extend({
         return this._super.apply(this, arguments);
     },
 
+    /**
+     * Check if the user is public, if it's true send a warning alert saying the action cannot be performed.
+     **/
+    _isUserPublic: function() {
```
Comment (tde-banana-odoo) [inline addons/website_forum/static/src/js/website_forum.js:193]: Maybe "_warnIfPublicUser" ?

**Review by tde-banana-odoo — APPROVED** [review]
Small naming proposal otherwise good little cleaning :)

**Comment by amdi-odoo** [comment]
Hello @tde-banana-odoo :slightly_smiling_face: 
Updated ! :D
Thanks!

**Comment by tde-banana-odoo** [comment]
@robodoo override=ci/security

Reason: usage of markup, with controller input

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**